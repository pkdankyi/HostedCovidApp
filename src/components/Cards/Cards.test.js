import React from 'react'
import ReactDOM from 'react-dom'
import '@testing-library/jest-dom/extend-expect'
import { render, cleanup } from '@testing-library/react'

import Cards from './Cards'

afterEach(cleanup)

it('renders', () => {
    const { asFragment } = render(<Cards data={{confirmed: '/[0-9]/', recovered: '/[0-9]/', deaths: '/[0-9]/', lastUpdate:'/[0-9]/'}}/>)
    expect(asFragment()).toMatchSnapshot()
}) 

it('inserts text in ', () => {
    const { getByTestId, getByText } = render(<Cards data={{confirmed: '/[0-9]/', recovered: '/[0-9]/', deaths: '/[0-9]/', lastUpdate:'/[0-9]/'}} />)
    expect(getByTestId('typo_infected')).toHaveTextContent('Infected')
})

it('inserts text in ', () => {
    const { getByTestId, getByText } = render(<Cards data={{confirmed: '/[0-9]/', recovered: '/[0-9]/', deaths: '/[0-9]/', lastUpdate:'/[0-9]/'}} />)
    expect(getByTestId('typo_recovered')).toHaveTextContent('Recovered')
})

it('inserts text in ', () => {
    const { getByTestId, getByText } = render(<Cards data={{confirmed: '/[0-9]/', recovered: '/[0-9]/', deaths: '/[0-9]/', lastUpdate:'/[0-9]/'}} />)
    expect(getByTestId('typo_deaths')).toHaveTextContent('Deaths')
})

it('inserts text in ', () => {
    const { getByTestId, getByText } = render(<Cards data={{confirmed: '/[0-9]/', recovered: '/[0-9]/', deaths: '/[0-9]/', lastUpdate:'/[0-9]/'}} />)
    expect(getByTestId('typo_active')).toHaveTextContent('Active')
})



















// import React from 'react'
// import ReactDOM from 'react-dom'
// import { render, cleanup } from '@testing-library/react'
// // import "jest-dom/extend-expect"

// import Cards from './Cards'
// // import { createRenderer } from 'react-dom/test-utils'
// // import renderer from 'react-test-renderer'

// import { shallow } from '@testing-library/jest-dom'

// // afterEach(cleanup);
// describe('Cards', () => {
//     it('renders confirmed with text', () => {
//         const data = [{ text: 'foo'}]
//         const { getByText } = render(<Cards data={data} />)

//         getByText(data[0].text)
//     })
// })
// // it("renders without crashing", () => {
// //     const div = document.createElement('div');
// //     ReactDOM.render(<Cards />, div);
// // })
