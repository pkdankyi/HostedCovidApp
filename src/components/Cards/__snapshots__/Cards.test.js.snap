// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div
    class="container"
    data-testid="cards"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 MuiGrid-justify-xs-center"
    >
      <div
        class="MuiPaper-root MuiCard-root MuiGrid-root card infected MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-2 MuiPaper-elevation1 MuiPaper-rounded"
      >
        <div
          class="MuiCardContent-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-gutterBottom"
            data-testid="typo_infected"
          >
            Infected
          </p>
          <h5
            class="MuiTypography-root MuiTypography-h5"
          >
            <span />
          </h5>
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary"
          >
             Invalid Date 
          </p>
          <p
            class="MuiTypography-root MuiTypography-body2"
          >
            Number of confirmed cases of COVID-19
          </p>
        </div>
      </div>
      <div
        class="MuiPaper-root MuiCard-root MuiGrid-root card recovered MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-2 MuiPaper-elevation1 MuiPaper-rounded"
      >
        <div
          class="MuiCardContent-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-gutterBottom"
            data-testid="typo_recovered"
          >
            Recovered
          </p>
          <h5
            class="MuiTypography-root MuiTypography-h5"
          >
            <span />
          </h5>
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary"
          >
             Invalid Date 
          </p>
          <p
            class="MuiTypography-root MuiTypography-body2"
          >
            Number of recoveries from COVID-19
          </p>
        </div>
      </div>
      <div
        class="MuiPaper-root MuiCard-root MuiGrid-root card deaths MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-2 MuiPaper-elevation1 MuiPaper-rounded"
      >
        <div
          class="MuiCardContent-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-gutterBottom"
            data-testid="typo_deaths"
          >
            Deaths
          </p>
          <h5
            class="MuiTypography-root MuiTypography-h5"
          >
            <span />
          </h5>
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary"
          >
             Invalid Date 
          </p>
          <p
            class="MuiTypography-root MuiTypography-body2"
          >
            Number of deaths caused by COVID-19
          </p>
        </div>
      </div>
      <div
        class="MuiPaper-root MuiCard-root MuiGrid-root card active MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-2 MuiPaper-elevation1 MuiPaper-rounded"
      >
        <div
          class="MuiCardContent-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-gutterBottom"
            data-testid="typo_active"
          >
            Active
          </p>
          <h5
            class="MuiTypography-root MuiTypography-h5"
          >
            <span />
          </h5>
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary"
          >
             Invalid Date 
          </p>
          <p
            class="MuiTypography-root MuiTypography-body2"
          >
            Number of existing cases of COVID-19
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
